File: java/resistor-color-duo/src/main/java/ResistorColorDuo.java
import java.util.stream.Stream;

class ResistorColorDuo {
    enum Colors {BLACK, BROWN, RED, ORANGE, YELLOW, GREEN, BLUE, VIOLET, GREY, WHITE}

    int value(String[] colors) {
        return Stream.of(colors)
                .limit(2)
                .map(String::toUpperCase)
                .map(Colors::valueOf)
                .map(Colors::ordinal)
                .reduce((a, b) -> a * 10 + b)
                .orElse(0);
    }
}

File: c/resistor-color-duo/src/resistor_color_duo.c
#include "resistor_color_duo.h"

int color_code(resistor_band_t *color) {
    return (int) *color * 10 + color[1];
}


const resistor_band_t* colors() {
    static resistor_band_t result[] = {
            BLACK, BROWN, RED, ORANGE, YELLOW,
            GREEN, BLUE, VIOLET, GREY, WHITE
    };

    return result;
}